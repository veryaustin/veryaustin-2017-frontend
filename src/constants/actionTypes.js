export const DATA_FETCH = 'DATA_FETCH';