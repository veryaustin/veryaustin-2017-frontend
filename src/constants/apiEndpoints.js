export const DATA_ROOT_URL = "https://veryaustin-55602.firebaseio.com";
export const STATIC_ASSETS_URL = "http://static.veryaustin.com";
