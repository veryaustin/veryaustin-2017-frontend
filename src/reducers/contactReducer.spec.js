import * as types from '../constants/actionTypes';
import reducer from './contactReducer';

describe('Contact Reducer',() => {
  it('should return the initial state', () => {
    const action = {type: 'unknown'};
    expect(reducer(undefined, action)).toMatchSnapshot();
  });

  it('should handle the CONTACT_FETCH action', () => {
    const initialState = {};

    const testData = {
      title: 'Test Contact Title',
      slogan: 'Test Contact Slogan',
      contactInfo: [
        {
          title: 'contact_1',
          icon_url: 'http://test_contact_1.com',
          site_url: 'test@contact_1.com',
        },
        {
          title: 'contact_2',
          icon_url: 'http://test_contact_2.com',
          site_url: 'test@contact_2.com',
        }
      ],
    };

    const action = {
      type: types.CONTACT_FETCH,
      payload: {data: testData}
    };
    
    expect(reducer(initialState, action)).toMatchSnapshot();
  });
});