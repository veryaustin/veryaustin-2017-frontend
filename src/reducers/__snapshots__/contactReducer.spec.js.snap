exports[`Contact Reducer should handle the CONTACT_FETCH action 1`] = `
Object {
  "contactInfo": Array [
    Object {
      "icon_url": "http://test_contact_1.com",
      "site_url": "test@contact_1.com",
      "title": "contact_1",
    },
    Object {
      "icon_url": "http://test_contact_2.com",
      "site_url": "test@contact_2.com",
      "title": "contact_2",
    },
  ],
  "slogan": "Test Contact Slogan",
  "title": "Test Contact Title",
}
`;

exports[`Contact Reducer should return the initial state 1`] = `Object {}`;
